IOS10-HW11-Maria Glushtsova/Sources/ViewController.swift

```diff
@@ -0,0 +1,331 @@
+        button.backgroundColor = UIColor(named: "green")
+        button.layer.cornerRadius = 20
+        button.layer.shadowColor = UIColor.black.cgColor
+        button.layer.shadowOpacity = 0.3
+        button.layer.shadowOffset = .zero
+        button.layer.shadowRadius = 10
+        button.layer.shouldRasterize = true
+        button.layer.rasterizationScale = UIScreen.main.scale
+        return button
+    }()
+    
+    private lazy var twitterButton: UIButton = {
+        let button = UIButton(type: .system)
+        button.addTarget(self, action: #selector(pressedTwitter), for: .touchUpInside)
+        button.setTitle("Twitter", for: .normal)
+        button.setTitleColor(.black, for: .normal)
+        button.titleLabel?.font = UIFont.systemFont(ofSize: 13)
+        button.backgroundColor = UIColor(named: "green")
+        button.layer.cornerRadius = 20
+        button.layer.shadowColor = UIColor.black.cgColor
+        button.layer.shadowOpacity = 0.3
+        button.layer.shadowOffset = .zero
+        button.layer.shadowRadius = 10
+        button.layer.shouldRasterize = true
+        button.layer.rasterizationScale = UIScreen.main.scale
+        return button
+    }()
+    
+    private lazy var facebookImage: UIImageView = {
+        let image = UIImageView(image: UIImage(named: "facebook"))
+        return image
+    }()
+    
+    private lazy var twitterImage: UIImageView = {
+        let image = UIImageView(image: UIImage(named: "twitter"))
+        return image
+    }()
+    
+    private lazy var signUpLabel: UILabel = {
+        let label = UILabel()
+        label.text = "Dont have account?"
+        label.textColor = UIColor(named: "colorField")
+        label.font = UIFont.systemFont(ofSize: 14)
+        return label
+    }()
+    
+    private lazy var signUpButton: UIButton = {
+        let button = UIButton(type: .system)
+        button.addTarget(self, action: #selector(pressedSingUp), for: .touchUpInside)
+        button.setTitle("Sign Up", for: .normal)
+        button.setTitleColor(.black, for: .normal)
+        button.titleLabel?.font = UIFont.systemFont(ofSize: 14)
+        return button
+    }()
+    
+    // MARK: - Lifecycle
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupHierarchy()
+        setupLayout()
+    }
+    
+    // MARK: - Setup
+    
+    private func setupHierarchy() {
+        
+        view.addSubview(login)
+        view.addSubview(loginField)
+        view.addSubview(passwordField)
+        view.addSubview(loginButton)
+        view.addSubview(forgotPasswordButton)
+        view.addSubview(leftLine)
+        view.addSubview(connectWith)
+        view.addSubview(rightLine)
+        view.addSubview(facebookButton)
+        view.addSubview(twitterButton)
+        view.addSubview(facebookImage)
+        view.addSubview(twitterImage)
+        view.addSubview(signUpLabel)
+        view.addSubview(signUpButton)
+        
```

Comment:
отступы на 167 и после
+ это можно записать так
```suggestion
     [login, loginField, passwordField, loginButton,
         forgotPasswordButton, leftLine, connectWith,
         rightLine, facebookButton, twitterButton, facebookImage,
         twitterImage, signUpLabel, signUpButton].forEach {
            view.addSubview($0)
        }
```